docs/user-manuals/persistentpodstate.md

```diff
@@ -0,0 +1,156 @@
+Many of the stateful services migrated to K8S are not completely cloud-native transformation, and still retain many of the deployment features of the physical machine era, such as:
+
+- **Service discovery Middleware services are exceptionally sensitive to the Pod IP after deployment, requiring that the IP cannot be changed.**
+
+- **Database related services persist data to the host disk, and changes to the Node to which they belong will result in data loss.**
+
+For such stateful services as described above, which have a strong dependency on Pod deployment state, Kruise is able to maintain the state of Pods after their first deployment to some extent by PersistentPodState CRD, e.g., "Fixed IP Scheduling".
+
+## API Definition
+```yaml
+apiVersion: apps.kruise.io/v1alpha1
+kind: PersistentPodState
+metadata:
+  name: echoserver
+  namespace: echoserver
+spec:
+  targetRef:
+    # Native k8s or kruise StatefulSet
+    # only support StatefulSet
+    apiVersion: apps.kruise.io/v1beta1
+    kind: StatefulSet
+    name: echoserver
+  # required node affinity. As follows, Pod rebuild will force deployment to the same Zone
+  requiredPersistentTopology:
+    nodeTopologyKeys:
+      failure-domain.beta.kubernetes.io/zone[,other node labels]
+  # preferred node affinity. As follows, Pod rebuild will preferred deployment to the same node
+  preferredPersistentTopology:
+    - preference:
+        nodeTopologyKeys:
+          kubernetes.io/hostname[,other node labels]
+      weight: 100
+```
+
+## Implementation
+PersistentPodState implementation principle is as follows, for detailed design please refer to: [PPS Proposal](https://github.com/openkruise/kruise/blob/master/docs/proposals/20220421-persistent-pod-state.md).
+
+![PersistentPodState](/img/docs/user-manuals/persistent-pod-state.png)
+
+## Fixed IP Scheduling Practice
+"Fixed IP scheduling" should be a common requirement for K8S deployments of stateful services. It does not mean "Specified Pod IP", but requires that the Pod IP does not change after the first deployment, either by service release or by machine eviction.
+To achieve this, we need the K8S network component to support Pod IP retention and the ability to keep the IP as unchanged as possible.
+In this article, we have modified the Host-local plugin in the flannel network component so that it can achieve the effect of keeping the Pod IP unchanged under the same Node.
+Related principles will not be stated here, please refer to the code: [host-local](https://github.com/openkruise/samples/tree/master/containernetworking/plugins).
+
+Here we have some questions, "Fixed IP scheduling" seems to be supported by the network component, what does this have to do with PersistentPodState?
```

Comment:
IP retention seems to be supported by the network component, how does it related with PersistentPodState?